Formula/elm.rb

```diff
@@ -18,10 +18,14 @@ class Elm < Formula
   uses_from_macos "ncurses"
   uses_from_macos "zlib"
 
-  def install
-    # elm-compiler needs to be staged in a subdirectory for the build process to succeed
-    (buildpath/"elm-compiler").install Dir["*"]
+  patch do
+    # elm's tarball is not a proper cabal tarball, it contains multiple cabal files.
+    # Add `cabal.project` lets cabal-install treat this tarball as cabal project correctly.
+    url "https://github.com/fishtreesugar/compiler/commit/eb566e901a419a6620e43c18faf89f57f0827124.patch?full_index=1"
```

Comment:
```suggestion
    # https://github.com/elm/compiler/pull/<insert PR number here>
    url "https://github.com/elm/compiler/commit/eb566e901a419a6620e43c18faf89f57f0827124.patch?full_index=1"
```

Reply:
Nice work! Could you submit this as a PR to upstream?

Reply:
done